### PR TITLE
Pass through a data option to the model fetch.

### PR DESCRIPTION
**Pull request by manthey**
Document the fetch options.

This adds unit tests for some of the base Model class.

### CODE REVIEW

File: clients/web/test/spec/modelSpec.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            } else {
+                resp.resolve({});
+            }
+            // The jqXHR response of a rest request needs an error function
+            resp.error = resp.fail;
+            return resp;
+        });
+
+        windowAlert = window.alert;
+        window.alert = function (msg) {
```
Comment (zachmullen) [inline clients/web/test/spec/modelSpec.js]: Why do we need to mock alert here?

Comment (manthey) [inline clients/web/test/spec/modelSpec.js]: It is just a way to get a counter to see if we called it.

Comment (zachmullen) [inline clients/web/test/spec/modelSpec.js]: IIRC, calling `alert` in phantom (at least in older versions) would hang the test indefinitely :)

Comment (zachmullen) [inline clients/web/test/spec/modelSpec.js]: Hence, maybe we should remove the `alert` from our model code in favor of a `console.error`.

Comment (manthey) [inline clients/web/test/spec/modelSpec.js]: Yes.  So in the tests, we normally just mock `window.alert` and have it call `console.log` to prevent that.

Comment (manthey) [inline clients/web/test/spec/modelSpec.js]: We use alert in a number of places: Collection, AccessControlledModel, GroupView, and Model.  All of them are really for a failure to set up a model properly (thus to inform the developer and not the user).  Converting them to `console.error` seems fine to me.  Would you like that on this or a different PR?

Comment (manthey) [inline clients/web/test/spec/modelSpec.js]: The alert in the GroupView has a note to `don't alert, show something useful`.

Comment (zachmullen) [inline clients/web/test/spec/modelSpec.js]: Yeah we can do the alert squashing on a separate PR, and at the same time add something in the test env to mock it out and cause it to fail the test if it's called.
